src/datasets/load.py

```diff
@@ -771,10 +775,14 @@ class PackagedDatasetModuleFactory(_DatasetModuleFactory):
     def __init__(
         self,
         name: str,
+        data_dir: Optional[str] = None,
         data_files: Optional[Union[str, List, Dict]] = None,
         download_config: Optional[DownloadConfig] = None,
         download_mode: Optional[DownloadMode] = None,
     ):
+        if data_files is None and data_dir is not None:
+            data_files = [os.path.join(data_dir, "*"), os.path.join(data_dir, "**/*")]
```

Comment:
Does this mean that this code would only return a split "train" ?
```python
# data/ contains two files train.json and test.json
load_dataset("json", data_dir="data")  # returns only train ?
```

It can be confusing given that
```python
load_dataset("data")  # returns train and test
```

On the other hand it is normal that
```python
load_dataset("json", data_files="data/**")  # returns only train
# because it is equivalent to
load_dataset("json", data_files={"train": "data/**"})  # returns only train
```

Do you think this is fine or we need a change ?

I can imagine that some dataset builders based on standard structures would like to return several splits depending on what's inside `data_dir`

Reply:
Hmm, this feature's goal is to be a shorthand syntax for `f"{data_dir}/**"`, and I'm not sure it is safe to assume that the user wants additional `data_dir` processing (automatic split globbing as we do with `data_files`, for instance) on top of that, so I'd leave this as-is. This can be changed later, depending on the feedback we get.

Reply:
Ok let's keep it this way then :)